sdcm/nemesis.py

```diff
@@ -434,7 +438,123 @@ def disrupt_stop_start_scylla_server(self):  # pylint: disable=invalid-name
+
+    def disrupt_start_stop_validation_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 120
+        trigger_func = partial(compaction_ops.trigger_validation_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             watch_for="Scrubbing in validate mode",
+                             timeout=timeout,
+                             stop_func=compaction_ops.stop_validation_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_upgrade_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 300
+        upgraded_configuration_options = {"enable_sstables_mc_format": False,
+                                          "enable_sstables_md_format": True}
+
+        def _upgrade_sstables_format(node: BaseNode):
+            LOGGER.info("Upgrading sstables format...")
+            with node.remote_scylla_yaml() as scylla_yaml:
+                scylla_yaml.update(upgraded_configuration_options)
+
+        node.run_nodetool("flush")
+        node.stop_scylla()
+        _upgrade_sstables_format(node)
+        node.start_scylla()
+
+        trigger_func = partial(compaction_ops.trigger_upgrade_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             watch_for="Upgrade keyspace1.standard1",
+                             timeout=timeout,
+                             stop_func=compaction_ops.stop_upgrade_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_reshape_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 300
+
+        def _trigger_reshape(node: BaseNode, tester: ClusterTester, keyspace: str = "keyspace1"):
+            twcs = {'class': 'TimeWindowCompactionStrategy', 'compaction_window_size': 1,
+                    'compaction_window_unit': 'MINUTES', 'max_threshold': 1, 'min_threshold': 1}
+            node.run_nodetool(sub_cmd="flush")
+            tester.wait_no_compactions_running()
+            LOGGER.info("Copying data files to ./staging and ./upload directories...")
+            keyspace_dir = f'/var/lib/scylla/data/{keyspace}'
+            cf_data_dir = node.remoter.run(f"ls {keyspace_dir}").stdout.splitlines()[0]
+            full_dir_path = f"{keyspace_dir}/{cf_data_dir}"
+            upload_dir = f"{full_dir_path}/upload"
+            staging_dir = f"{full_dir_path}/staging"
+            cp_cmd_upload = f"cp -p {full_dir_path}/mc-* {upload_dir}"
```

Comment:
need to make it general enough

Reply:
Fixed (made to be cover both sstable file format prefixes).